### ISSUE TITLE
[Tutorial] Collaboration Guidelines with All Contributors

### ISSUE THREAD

**Issue by zhengh96** [dptech-corp/Uni-Dock-Dev#8]
We are now practicing a collaboration standard, which is to first list the required features through **Github Issues**, then develop them, and associate the **Pull Requests** that have completed the development to the corresponding Issues. 
We expect this specification to become a consensus for Uni-Dock contributors, please indicate it in the README.

**Comment by zhengh96** [comment]
I DO think the open source collaboration in the main Uni-Dock repo is the most appropriate solution, but now it seems a bit abrupt to collaborate directly on the main repo in a new organizational form (the Issue driven form). 

So we first use the Uni-Dock-Dev repo to complete the development of Uni-Dock v1.1 and the refactor-like issues, and add the collaboration standard to the README (as described in this issue). 

Then we switch to the Uni-Dock main repo for collaboration.

**Comment by caic99** [comment]
> I DO think the open source collaboration in the main Uni-Dock repo is the most appropriate solution, but now it seems a bit abrupt to collaborate directly on the main repo in a new organizational form (the Issue driven form). 
> 
> 
> 
> So we first use the Uni-Dock-Dev repo to complete the development of Uni-Dock v1.1 and the refactor-like issues, and add the collaboration standard to the README (as described in this issue). 
> 
> 
> 
> Then we switch to the Uni-Dock main repo for collaboration.

Agreed. I will draft a contribution guideline following the open source best practices.

**Comment by zhengh96** [comment]
I think we could also plan the design of the content to be displayed in the README at the root level of the Uni-Dock Repo, such as a basic introduction, collaboration guidelines, recent updates, and so on.

Any ideas? @ysyecust @caic99 @YNYuan

**Comment by zhengh96** [comment]
We may follow this [recommendation](https://github.com/dptech-corp/Uni-Dock/community).

**Comment by zhengh96** [comment]
Two ideas:
- Add title with issue type automatically
- Optimize the context template

```markdown
[Feature Request]
> A clear and concise description of what the problem is.
[Possible Solution (Optional)]
> A clear and concise description of what you want to happen.
[Alternative Solution (Optional)]
> A clear and concise description of any alternative solutions or features you've considered.
[Additional Context (Optional)]
> Add any other context or screenshots about the feature request here.
```

**Comment by caic99** [comment]
> Two ideas:
> 
> * Add title with issue type automatically
> * Optimize the context template

Implemented in #34 using issue forms.
